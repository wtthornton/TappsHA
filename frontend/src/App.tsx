import { QueryProvider } from './providers/QueryProvider';
import { AuthProvider, ProtectedRoute } from './contexts/AuthContext';
import ConnectionStatusDashboard from './components/ConnectionStatusDashboard';
import EventMonitoringDashboard from './components/EventMonitoringDashboard';
import HomeAssistantConnectionForm from './components/HomeAssistantConnectionForm';
import LoginForm from './components/LoginForm';

// âœ… Context7-validated React 19 functional component pattern
// Following React 19 best practices from Context7 documentation
function App() {
  return (
    <AuthProvider>
      <QueryProvider>
        <div className="min-h-screen bg-gray-50">
          <header className="bg-white shadow-sm border-b border-gray-200">
            <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
              <div className="flex justify-between items-center py-6">
                <div className="flex items-center">
                  <h1 className="text-2xl font-bold text-gray-900">TappHA</h1>
                  <span className="ml-2 text-sm text-gray-500">Home Assistant Integration</span>
                </div>
                <nav className="flex space-x-8">
                  <a href="/dashboard" className="text-gray-500 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">
                    Dashboard
                  </a>
                  <a href="/connections" className="text-gray-500 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">
                    Connections
                  </a>
                  <a href="/events" className="text-gray-500 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium">
                    Events
                  </a>
                </nav>
              </div>
            </div>
          </header>

          <main className="max-w-7xl mx-auto py-6 sm:px-6 lg:px-8">
            {/* Protected Routes - Using Context7 React 19 pattern */}
            <ProtectedRoute>
              <div className="space-y-6">
                {/* Connection Management Section */}
                <section className="bg-white shadow rounded-lg p-6">
                  <h2 className="text-lg font-medium text-gray-900 mb-4">Home Assistant Connection</h2>
                  <HomeAssistantConnectionForm />
                </section>

                {/* Connection Status Dashboard */}
                <section className="bg-white shadow rounded-lg p-6">
                  <h2 className="text-lg font-medium text-gray-900 mb-4">Connection Status</h2>
                  <ConnectionStatusDashboard />
                </section>

                {/* Event Monitoring Dashboard */}
                <section className="bg-white shadow rounded-lg p-6">
                  <h2 className="text-lg font-medium text-gray-900 mb-4">Event Monitoring</h2>
                  <EventMonitoringDashboard />
                </section>
              </div>
            </ProtectedRoute>

            {/* Public Routes */}
            <div id="login-route" style={{ display: 'none' }}>
              <LoginForm />
            </div>
          </main>
        </div>
      </QueryProvider>
    </AuthProvider>
  );
}

export default App;
