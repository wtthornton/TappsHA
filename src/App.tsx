import { Routes, Route } from 'react-router-dom'
import { Toaster } from '@/components/ui/toaster'
import Layout from '@/components/Layout'
import HomePage from '@/pages/HomePage'
import AboutPage from '@/pages/AboutPage'

function App() {
  return (
    <div className="min-h-screen bg-background">
      <Layout>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/about" element={<AboutPage />} />
        </Routes>
      </Layout>
      <Toaster />
    </div>
  )
}

export default App 