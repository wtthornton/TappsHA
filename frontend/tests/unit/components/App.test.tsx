import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import App from '../../../src/App';

// Mock the API client
vi.mock('../../../src/services/api', () => ({
  apiClient: {
    get: vi.fn(),
    post: vi.fn(),
  },
}));

// Create a test query client
const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

describe('App Component', () => {
  it('should render without crashing', () => {
    const queryClient = createTestQueryClient();
    
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <App />
      </QueryClientProvider>
    );
    
    expect(container).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const queryClient = createTestQueryClient();
    
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <App />
      </QueryClientProvider>
    );
    
    expect(container).toMatchSnapshot();
  });

  it('should display main navigation', () => {
    const queryClient = createTestQueryClient();
    
    render(
      <QueryClientProvider client={queryClient}>
        <App />
      </QueryClientProvider>
    );
    
    // Check for navigation elements
    const nav = screen.getByRole('navigation');
    expect(nav).toBeInTheDocument();
  });

  it('should display main content area', () => {
    const queryClient = createTestQueryClient();
    
    render(
      <QueryClientProvider client={queryClient}>
        <App />
      </QueryClientProvider>
    );
    
    // Check for main content
    const main = screen.getByRole('main');
    expect(main).toBeInTheDocument();
  });

  it('should handle loading state', () => {
    const queryClient = createTestQueryClient();
    
    render(
      <QueryClientProvider client={queryClient}>
        <App />
      </QueryClientProvider>
    );
    
    // Check for loading indicators
    const loadingElements = screen.queryAllByText(/loading/i);
    expect(loadingElements.length).toBeGreaterThanOrEqual(0);
  });

  it('should handle error state', () => {
    const queryClient = createTestQueryClient();
    
    render(
      <QueryClientProvider client={queryClient}>
        <App />
      </QueryClientProvider>
    );
    
    // Check for error handling
    const errorElements = screen.queryAllByText(/error/i);
    expect(errorElements.length).toBeGreaterThanOrEqual(0);
  });
}); 