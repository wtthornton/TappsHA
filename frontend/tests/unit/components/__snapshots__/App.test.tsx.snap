// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > should match snapshot 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-50"
  >
    <header
      class="bg-white shadow-sm border-b border-gray-200"
    >
      <div
        class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
      >
        <div
          class="flex justify-between items-center py-6"
        >
          <div
            class="flex items-center"
          >
            <h1
              class="text-2xl font-bold text-gray-900"
            >
              TappHA
            </h1>
            <span
              class="ml-2 text-sm text-gray-500"
            >
              Home Assistant Integration
            </span>
          </div>
          <nav
            class="flex space-x-8"
          >
            <a
              class="text-gray-500 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium"
              href="/dashboard"
            >
              Dashboard
            </a>
            <a
              class="text-gray-500 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium"
              href="/connections"
            >
              Connections
            </a>
            <a
              class="text-gray-500 hover:text-gray-700 px-3 py-2 rounded-md text-sm font-medium"
              href="/events"
            >
              Events
            </a>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="max-w-7xl mx-auto py-6 sm:px-6 lg:px-8"
    >
      <div
        class="flex items-center justify-center min-h-screen"
      >
        <div
          class="text-center"
        >
          <h2
            class="text-2xl font-bold text-gray-900 mb-4"
          >
            Authentication Required
          </h2>
          <p
            class="text-gray-600 mb-4"
          >
            Please log in to access this page.
          </p>
          <button
            class="btn-primary"
          >
            Go to Login
          </button>
        </div>
      </div>
      <div
        id="login-route"
        style="display: none;"
      >
        <div
          class="min-h-screen flex items-center justify-center bg-gray-50 py-12 px-4 sm:px-6 lg:px-8"
        >
          <div
            class="max-w-md w-full space-y-8"
          >
            <div>
              <h2
                class="mt-6 text-center text-3xl font-extrabold text-gray-900"
              >
                Sign in to TappHA
              </h2>
              <p
                class="mt-2 text-center text-sm text-gray-600"
              >
                Home Assistant Integration Platform
              </p>
            </div>
            <form
              class="mt-8 space-y-6"
            >
              <div
                class="space-y-4"
              >
                <div>
                  <label
                    class="block text-sm font-medium text-gray-700"
                    for="username"
                  >
                    Username
                  </label>
                  <input
                    autocomplete="username"
                    class="mt-1 block w-full px-3 py-2 border rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-primary-500 focus:border-primary-500 sm:text-sm border-gray-300"
                    id="username"
                    name="username"
                    placeholder="Enter your username"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <div>
                  <label
                    class="block text-sm font-medium text-gray-700"
                    for="password"
                  >
                    Password
                  </label>
                  <input
                    autocomplete="current-password"
                    class="mt-1 block w-full px-3 py-2 border rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-primary-500 focus:border-primary-500 sm:text-sm border-gray-300"
                    id="password"
                    name="password"
                    placeholder="Enter your password"
                    required=""
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div>
                <button
                  class="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 bg-primary-600 hover:bg-primary-700"
                  type="submit"
                >
                  Sign in
                </button>
              </div>
              <div
                class="flex items-center justify-between"
              >
                <div
                  class="text-sm"
                >
                  <a
                    class="font-medium text-primary-600 hover:text-primary-500"
                    href="/forgot-password"
                  >
                    Forgot your password?
                  </a>
                </div>
                <div
                  class="text-sm"
                >
                  <a
                    class="font-medium text-primary-600 hover:text-primary-500"
                    href="/register"
                  >
                    Don't have an account?
                  </a>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
